pkg/storage/edv/restprovider.go

```diff
@@ -213,70 +228,171 @@ func (r *restStore) Get(k string) ([]byte, error) {
 	return encryptedDocumentBytes, nil
 }
 
-// Not support by restStore. The EDV data vault HTTPS API doesn't support this type of operation,
-// nor is there a feasible way to emulate it. The iterator returned always is always in an error state.
-func (r *restStore) Iterator(string, string) storage.StoreIterator {
-	return &restStoreIterator{}
+// Iterator returns all documents within the store. It does not support start and end key filtering.
+func (r *restStore) Iterator(_, _ string) storage.StoreIterator {
+	allDocumentLocations, err := r.getAllDocumentLocations()
+	if err != nil {
+		return &restStoreIterator{err: fmt.Errorf(failGetAllDocumentLocations, err)}
+	}
+
+	allDocuments, err := r.getAllDocuments(allDocumentLocations)
```

Comment:
I don't know what the typical number of documents this will return. If it's a small number then it's fine to load all documents into memory. If it turns out to be in the order of hundreds/thousands then I would rework this logic to possibly load all of the document IDs into memory and then retrieve each document on demand during iteration.

Reply:
@bstasyszyn this solution will change @DRK3 create task for refactor the interface

Reply:
@bstasyszyn @fqutishat Here's the task: #2315 

I also added a comment to address @bstasyszyn's comment in that issue.